### PR TITLE
cargo-udeps 0.1.30 (new formula)

### PR DESCRIPTION
**Pull request by rouge8**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/cargo-udeps.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        name = "demo-crate"
+        version = "0.1.0"
+
+        [dependencies]
+        clap = "3"
+      EOS
+
+      output = shell_output("cargo udeps 2>&1", 101)
+      # `cargo udeps` can be installed on Rust stable, but only runs with cargo with `cargo +nightly udeps`
+      assert_match "error: the option `Z` is only accepted on the nightly compiler", output
```
Comment (rouge8) [inline Formula/cargo-udeps.rb:29]: This is kind of weird, but I couldn't think of a better test...

File: Formula/cargo-udeps.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class CargoUdeps < Formula
+  desc "Find unused dependencies in Cargo.toml"
+  homepage "https://github.com/est31/cargo-udeps"
+  url "https://github.com/est31/cargo-udeps/archive/refs/tags/v0.1.30.tar.gz"
+  sha256 "a1dd8b533fa915783919b78705d4a377fb021b67f8386023866017a139caf935"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  depends_on "rust" => :build
```
Comment (carlocab) [inline Formula/cargo-udeps.rb:8]:
```suggestion
  depends_on "rust" => [:build, :test]
```

**Review by carlocab — DISMISSED** [review]

File: Formula/cargo-udeps.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class CargoUdeps < Formula
+  desc "Find unused dependencies in Cargo.toml"
+  homepage "https://github.com/est31/cargo-udeps"
+  url "https://github.com/est31/cargo-udeps/archive/refs/tags/v0.1.30.tar.gz"
+  sha256 "a1dd8b533fa915783919b78705d4a377fb021b67f8386023866017a139caf935"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  depends_on "rust" => [:build, :test]
+
+  def install
```
Comment (carlocab) [inline Formula/cargo-udeps.rb:10]:
```suggestion
  depends_on "rust" =>  [:build, :test]

  on_linux do
    depends_on "pkg-config" => :build
    depends_on "openssl@1.1"
  end

  def install
```

File: Formula/cargo-udeps.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class CargoUdeps < Formula
+  desc "Find unused dependencies in Cargo.toml"
+  homepage "https://github.com/est31/cargo-udeps"
+  url "https://github.com/est31/cargo-udeps/archive/refs/tags/v0.1.30.tar.gz"
+  sha256 "a1dd8b533fa915783919b78705d4a377fb021b67f8386023866017a139caf935"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  depends_on "rust" =>  [:build, :test]
```
Comment (carlocab) [inline Formula/cargo-udeps.rb:8]:
```suggestion
  depends_on "rust" => [:build, :test]
```
Sorry, accidentally added a space in my last suggestion

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**